adapters/Redis/RedisMemoryStorage/RedisConfig.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.KernelMemory.Redis;
+
+/// <summary>
+/// Lays out the tag fields that you want redis to index.
+/// </summary>
+public class RedisConfig
+{
+    /// <summary>
+    /// Gets or sets the Redis connection string based on
+    /// <see href="https://stackexchange.github.io/StackExchange.Redis/Configuration">StackExchange.Redis' connection string</see>
+    /// </summary>
+    public string ConnectionString { get; set; }
+
+    /// <summary>
+    /// The Collection of tags that you want to be able to search on.
+    /// The Key is the tag name, and the char is the separator that you
+    /// want Redis to use to separate your tag fields. The default separator
+    /// is ','.
+    /// </summary>
+    public Dictionary<string, char?> Tags { get; } = new()
+    {
+        { Constants.ReservedDocumentIdTag, '|' },
+        { Constants.ReservedFileIdTag, '|' },
+        { Constants.ReservedFilePartitionTag, '|' },
+        { Constants.ReservedFileTypeTag, '|' },
+    };
+
+    /// <summary>
+    /// Initializes an instance of RedisMemoryConfiguration.
+    /// </summary>
+    /// <param name="tags">The collection of tags you want to be able to search on. The key</param>
+    public RedisConfig(IEnumerable<KeyValuePair<string, char?>>? tags = null)
+    {
+        if (tags is not null)
+        {
+            foreach (var tag in tags)
+            {
+                if (!this.Tags.TryAdd(tag.Key, tag.Value))
+                {
+                    this.Tags[tag.Key] = tag.Value;
+                }
```

Comment:
Maybe we could just use `Tags[tag.Key] = tag.Value`

Reply:
why is it a dictionary rather than a list of tag names?

Reply:
@WeihanLi - good point not sure why I didn't do that in the first place

@dluc - due to the Semantics of a tag field in Redis. Tag fields are separated by a separator character (default is a comma)

Hence you could have blog.tag tech,ai,redis,azure and match on tech, ai, redis, or azure. The dictionary is to give the user the flexibility to use fields with those tag separators in them. Might be worth adding a constructor with a list that just defaults everything to a comma.